File: Convert 1D Array Into 2D Array/code.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& nums, int n, int m) 
    {
        if(n*m!=nums.size())
        {
            return {};
        }
        int curr=0;
        vector<vector<int>>v(n,vector<int>(m,0));
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                v[i][j]=nums[curr++];
            }
        }
        return v;
    }
};

File: Product of Array Except Self/code.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int nz=0;//number of zeros
        int prod=1;//product
        int n=nums.length;
        int[] ans=new int[n];
        for(int i=0;i<n;i++)
        {
            if(nums[i]==0)
                nz++;
            else
                prod=prod*nums[i];
        }
        for(int i=0;i<n;i++)
        {
            if(nz==1)
            {
                if(nums[i]==0)
                    ans[i]=prod;
                else
                    ans[i]=0;
            }
            if(nz==0)
                ans[i]=prod/nums[i];
        }
        return ans;
    }
}